controllers/custom/builder.go

```diff
@@ -160,6 +166,7 @@ func (b *Builder) Complete(reconciler Reconciler) (healthz.Checker, error) {
 
 				case cache.Deleted:
```

Comment:
I feel we need to track every delete event that is reconciled. We are not sure where was broken yet.